includes/Modules/Analytics_4.php

```diff
@@ -263,13 +289,52 @@ protected function create_data_request( Data_Request $data ) {
 	 * @return mixed Parsed response data on success, or WP_Error on failure.
 	 */
 	protected function parse_data_response( Data_Request $data, $response ) {
+		$filter_account = function( $account ) {
+			$matches = array();
+			if ( preg_match( '#accounts/([^/]+)#', $account['name'], $matches ) ) {
+				$account['_ID'] = $matches[1];
+			}
+
+			return $account;
+		};
+
+		$filter_property = function( $property ) {
+			$matches = array();
+			if ( preg_match( '#properties/([^/]+)#', $property['name'], $matches ) ) {
+				$property['_ID'] = $matches[1];
+			}
+
+			$matches = array();
+			if ( preg_match( '#accounts/([^/]+)#', $property['parent'], $matches ) ) {
+				$property['_accountID'] = $matches[1];
+			}
+
+			return $property;
+		};
+
+		$filter_webdatastream = function( $webdatastream ) {
+			$matches = array();
+			if ( preg_match( '#properties/([^/]+)/webDataStreams/([^/]+)#', $webdatastream['name'], $matches ) ) {
+				$webdatastream['_ID']         = $matches[2];
+				$webdatastream['_propertyID'] = $matches[1];
+			}
+
+			return $webdatastream;
+		};
```

Comment:
Same here, can we define private or protected methods for these three? E.g.:
* `filter_account_with_ids( $account )`
* `filter_property_with_ids( $property )`
* `filter_webdatastream_with_ids( $webdatastream )`

Reply:
Updated to be public static methods.